frame/system/src/offchain.rs

```diff
@@ -130,9 +162,70 @@ pub trait SubmitUnsignedTransaction<T: crate::Trait, Call> {
 	}
 }
 
+/// A utility trait to easily create signed transactions
+/// from accounts in node's local keystore.
+///
+/// NOTE: Most likely you should not implement this trait yourself.
+/// There is an implementation for `TransactionSubmitter` type, which
+/// you should use.
+pub trait SubmitSignedTransaction<T: crate::Trait, Call> {
+	/// A `SignAndSubmitTransaction` implementation.
+	type SignAndSubmit: SignAndSubmitTransaction<T, Call>;
+
+	/// Find local keys that match given list of accounts.
+	///
+	/// Technically it finds an intersection between given list of `AccountId`s
+	/// and accounts that are represented by public keys in local keystore.
+	///
+	/// Returns both public keys and `AccountId`s of accounts that are available.
+	/// Such accounts can later be used to sign a payload or send signed transactions.
+	fn find_local_keys(accounts: impl IntoIterator<Item = T::AccountId>) -> Vec<(
+		T::AccountId,
+		PublicOf<T, Call, Self::SignAndSubmit>,
+	)>;
+
+	/// Create and submit signed transactions from supported accounts.
+	///
+	/// This method should intersect given list of accounts with the ones
+	/// supported locally and submit signed transaction containing given `Call`
+	/// with every of them.
+	///
+	/// Returns a vector of results and account ids that were supported.
+	fn submit_signed(
+		accounts: impl IntoIterator<Item= T::AccountId>,
+		call: impl Into<Call> + Clone,
+	) -> Vec<(T::AccountId, Result<(), ()>)> {
```

Comment:
Do you think it makes sense to add an option to stop on the first successful submission?

Reply:
Why would that be useful? I think we should always submit with all accounts that are there (i.e. we do (or planned to) support mutliple validators on one node, right).

Reply:
In the equivocation report I was bailing out after the first success, but yeah probably makes sense to report from all available accounts anyway.